src/manifest/index.ts

```diff
@@ -18,7 +18,7 @@ const findManifestDependencies = async ({ manifest, isProduction, isStrict, dir,
 
   const scripts = Object.entries(manifest.scripts ?? {}).reduce((scripts, [scriptName, script]) => {
     if (script && (scriptFilter.length === 0 || scriptFilter.includes(scriptName))) {
```

Comment:
If you wanted to micro-optimise this for allocations a bit further, you could move `scriptFilter` out of the function and make it a set so that it has faster `.has()` checks, and also is only allocated once. Then rather than conditionally setting it to either `['start', 'posinstall']` or `[]`, move the `isProduction` check down here. So `scriptFilter` becomes `const scriptFilter = new Set(['start', 'postinstall']);` outside of the function, and this check becomes `if (script && (!isProduction || scriptFilter.has(scriptName))) {`

Reply:
I suggest making additions a bit later. I try not to make too many changes in the PR to reduce the likelihood of a bug. Moving the "isProduction" check is already a significant change, as it is already checked in another component. @webpro

Reply:
That's fine, but don't worry too much! We have test, e.g. the `isProduction` flag with the `start` has coverage.